src/Artsy/Router/makeAppRoutes.tsx

```diff
@@ -44,39 +45,50 @@ export function makeAppRoutes(routeList: RouteList[]): RouteConfig[] {
     createRouteConfiguration
   )
 
+  const Component = props => {
+    const { router, setRouter } = useSystemContext()
+    // Store global reference to router instance
+    useEffect(() => {
+      if (props.router !== router) {
+        setRouter(props.router)
+      }
+
+      /**
+       * Intercept <a> tags on page and if contained within router route
+       * manifest, navigate via router versus doing a hard jump between pages.
+       */
+      catchLinks(window, href => {
+        const url = ROUTE_NAMESPACE + href
+        const foundUrl = props.router.matcher.matchRoutes(routes, url)
+        // Make sure to track pageviews when staying on the same page type.
+        // Collection pages link to each other using `RouterLink` which already
+        // handles this, so just in case avoid double triggering.
+        if (foundUrl) {
+          const toPath = url.toString()
+          // TODO: Figure out why the router prop is stale.
+          // const currentPageType = props.match.location.pathname.split("/")[1]
+          const currentPageType = window.location.pathname.split("/")[1]
```

Comment:
I added an explicit `withRouter` HOC here (even tho it's probably not needed since this is a top-level route component anyway). Then, I tried to access the props in order to get the current page type, in order to accomplish this check.

However, I noticed something distressing! This seemed to be the original value, and not the current one. Consider the following flow:

you start on a collection page, click on an artwork -> routed to an artwork page, click on another artwork -> same artwork page, new params. In that case, `props.match` still contained the old `/collection` route (the original one from this flow).

<img width="703" alt="Screen Shot 2020-01-13 at 4 03 29 PM" src="https://user-images.githubusercontent.com/1457859/72293020-997a3d80-3620-11ea-9fba-0ebc6f9b9fd7.png">

Here's a pic, you can see we are clearly on an artwork page, being navigated to a different artwork. Yet our location prop here still refers to a collection page, which started our flow.

So, I opted to move to `window.`. Is there something with our router history I'm missing?

Reply:
When I'm back will take a look!